include/z64player.h

```diff
@@ -17,14 +17,14 @@ struct PlayState;
 #define PLAYER_GET_START_MODE(thisx) (((thisx)->params & 0xF00) >> 8)
 
 typedef enum PlayerStartMode {
-    /*  0x0 */ PLAYER_START_MODE_0,
-    /*  0x1 */ PLAYER_START_MODE_1, // Spawning after pulling/putting-back Master sword // OoT leftover
-    /*  0x2 */ PLAYER_START_MODE_2,
-    /*  0x3 */ PLAYER_START_MODE_3,
-    /*  0x4 */ PLAYER_START_MODE_4,
-    /*  0x5 */ PLAYER_START_MODE_5,
-    /*  0x6 */ PLAYER_START_MODE_6,
-    /*  0x7 */ PLAYER_START_MODE_7,
+    /*  0x0 */ PLAYER_START_MODE_NOTHING, // Update is empty and draw function is NULL, nothing occurs. Useful in cutscenes, for example.
+    /*  0x1 */ PLAYER_START_MODE_TIME_TRAVEL, // OoT Leftover. Arriving from time travel. Automatically adjusts by age.
+    /*  0x2 */ PLAYER_START_MODE_BLUE_WARP, // Arriving from a blue warp.
+    /*  0x3 */ PLAYER_START_MODE_DOOR, // Unused. Use a door immediately if one is nearby. If no door is in usable range, a softlock occurs.
+    /*  0x4 */ PLAYER_START_MODE_GROTTO, // Arriving from a grotto, launched upward from the ground.
+    /*  0x5 */ PLAYER_START_MODE_WARP_SONG, // OoT Leftover. Arriving from a warp song.
+    /*  0x6 */ PLAYER_START_MODE_OWL_STATUE, // Arriving at an owl statue, either from an owl save or an owl warp (song of soaring).
```

Comment:
Actually, before this is merged, it should be commented that `PLAYER_START_MODE_OWL_STATUE` includes the warp to the entrance of a dungeon by playing song of soaring inside that dungeon.

I'm wondering if `PLAYER_START_MODE_OWL_STATUE` should be a more general name to include that one case, since it is mostly for arriving at owl statues

Reply:
tbh for me, "owl statue" doesnt immediately imply "soar", which seems to be its more relevant (and common) use case

Reply:
I see 3 options:
`SOAR`: ignore owl saves for the sake of soaring
`OWL`: generally speaking this is all owl related stuff
`SOAR_AND_OWL_SAVE`/ `SOAR_AND_OWL_STATUE`: verbose for he sake of coverage

idk what is best

Reply:
I wanted to avoid "soar" because an owl save doesn't include song of soaring, and from my experiences, owl saves feel more common than warping to the entrance of a dungeon, but that might just be personal bias on how I played the game

Reply:
its still general soaring right, that in my point of view is more relevant and important than owl saves

Reply:
like, this mode is used for both regular soaring (to an owl), and soaring to a dungeon entrance. right?

Reply:
Yeah, it's three use cases are 

1) regular soaring (to an owl) where the start mode is in the params from the entrance given in `sOwlWarpEntrances` from `z_en_test7.c`

2) soaring to a dungeon entrance where the start mode is put directly into `gSaveContext.respawn[RESPAWN_MODE_TOP].playerParams` in `z_en_test7.c`

3) Loading the game from an owl save where the start mode is in the params from the entrance given in `sOwlWarpEntrances` from `z_sram_NES`

Maybe something like `PLAYER_START_MODE_OWL_SOARING` (or just `PLAYER_START_MODE_OWL_SOAR`) to cover all 3 cases?

Reply:
I feel like generally people will just interpret that as soaring. I think the only way to get coverage is to do `SOAR_AND_OWL_SAVE` or something similar. 
I dont have much of an opinion on what is chosen, maybe others can weigh in and help you choose

Reply:
TBH, I might just say `OWL`, since like Fig said this is all related to `OWL` entrances (while not technically not using owl statues, soaring to the entrance still acts like an OWL soar with song of soaring).
If we need more verbose I do kind of think you do need `AND` otherwise it does seem more just soaring.

Reply:
yeah even in the dungeon case it has a bunch of feather effects and the owl wings taking you :D it is certainly all owl stuff

Reply:
I went with just `OWL` and let the comments explain the rest.

I also named the action `Player_Action_OwlSaveArrive` so I'll be worth reapproving this PR before merging again

Reply:
Also noting `Player_Action_OwlSaveArrive` wasn't given a more general name specifically because `Message_StartTextbox(play, 0xC03, NULL);` starts the text box for arriving from an owl save. Everything else was just a generic cutscene action func